ssl/t1_lib.c

```diff
@@ -479,25 +479,29 @@ static int add_provider_sigalgs(const OSSL_PARAM params[], void *data)
     if (sinf->algorithm == NULL)
         goto err;
 
-    /* Truly optional */
+    /* Optional */
     p = OSSL_PARAM_locate_const(params, OSSL_CAPABILITY_TLS_SIGALG_HASHALG);
-    if (p == NULL || p->data_type != OSSL_PARAM_UTF8_STRING) {
+    if (p == NULL)
         sinf->hash_algorithm = NULL;
-    }
+    else if (p->data_type != OSSL_PARAM_UTF8_STRING)
+        goto err;
     else {
         sinf->hash_algorithm = OPENSSL_strdup(p->data);
         if (sinf->hash_algorithm == NULL)
             goto err;
     }
 
+    /* Optional */
     p = OSSL_PARAM_locate_const(params, OSSL_CAPABILITY_TLS_SIGALG_OID);
-    if (p == NULL || p->data_type != OSSL_PARAM_UTF8_STRING) {
-        ERR_raise(ERR_LIB_SSL, ERR_R_PASSED_INVALID_ARGUMENT);
+    if (p == NULL)
+        sinf->oid = NULL;
+    else if (p->data_type != OSSL_PARAM_UTF8_STRING)
```

Comment:
All branches should use `{}`

Reply:
Done in https://github.com/openssl/openssl/pull/19312/commits/8be6985a9743a239684308b6258691d3489e52bf

Reply:
Not done?

Reply:
Here, *all* of the `if` ... `else if` ... `else` bodies must be bracketed.

Reply:
done.